### PR TITLE
GO-3795 Add ObjectListModifyDetailValues RPC

### PR DESCRIPTION
**Pull request by KirillSto**
https://linear.app/anytype/issue/GO-3795/addremove-details-from-multiple-objects

New RPC request allows to set/add/remove detail values for multiple objects

### CODE REVIEW

File: util/slice/slice.go [hunk 1]
```diff
@@ -237,3 +237,24 @@ func FilterCID(cids []string) []string {
...
 }
+
+// MergeUniqBy merges two slices with comparator. Resulting slice saves values' order and uniqueness
+// Input slices MUST contain only unique values
+func MergeUniqBy[T comparable](s1, s2 []T, equal func(v1, v2 T) bool) (result []T) {
+	result = make([]T, len(s1))
+	copy(result, s1)
+	for _, v2 := range s2 {
+		isUnique := true
+		for _, v1 := range s1 {
```
Comment (deff7) [inline util/slice/slice.go:248]: you could use Contains

File: util/pbtypes/pbtypes.go [hunk 2]
```diff
@@ -582,3 +606,46 @@ func RelationIdToKey(id string) (string, error) {
 
 	return "", fmt.Errorf("incorrect id format")
 }
+
+// AddValue adds values to int lists and string lists
+func AddValue(s *types.Struct, key string, v *types.Value) {
```
Comment (deff7) [inline util/pbtypes/pbtypes.go:611]: I don't think these functions should be here, they are pretty specialized for only one task, so maybe move them close to place when they're called?

File: core/block/editor.go [hunk 3]
```diff
@@ -639,6 +639,36 @@ func (s *Service) ModifyDetails(objectId string, modifier func(current *types.St
...
+	var anySucceed bool
+	for _, objectId := range req.ObjectIds {
+		err := s.ModifyDetails(objectId, func(current *types.Struct) (*types.Struct, error) {
+			for _, op := range req.Operations {
+				if op.Set != nil && pbtypes.ValueToInterface(op.Set) != nil {
+					// Set operation has higher priority than Add and Remove, because it modifies full value
+					current.Fields[op.RelationKey] = op.Set
+					continue
+				}
+				pbtypes.AddValue(current, op.RelationKey, op.Add)
```
Comment (deff7) [inline core/block/editor.go:652]: maybe use switch / if between Set/Add/Remove operations? It will be clearer

File: util/pbtypes/pbtypes.go (hunk 2 above)
Comment (KirillSto) [reply util/pbtypes/pbtypes.go]: I moved it to separate file core/block/details.go, so all block.Service operations on details are described here

File: core/block/editor.go (hunk 3 above)
Comment (KirillSto) [reply core/block/editor.go]: I did not get it :see_no_evil: 
Can you please suggest how we can do it via switch

**Review by deff7 — APPROVED** [review]

**Pull request merged**